.github/workflows/dbld-images.yml

```diff
@@ -35,15 +29,16 @@ jobs:
           gh_export CONTAINER_REGISTRY
 
       - name: Build the images
-        run: dbld/rules images -j $(nproc)
+        run: dbld/rules images -j $(nproc) --keep-going --output-sync
 
       - name: Should we upload the images?
+        if: always()
         run: |
-          if [ "${{ github.event.inputs.upload_images }}" = "true" ] || \
+          if [ "${{ github.event.inputs.testing_image_upload }}" = "true" ] || \
              ( \
               [ "${{ github.repository_owner }}" = "syslog-ng" ] && \
               [ "${{ github.ref }}" = "refs/heads/master" ] && \
-              [ "${{ github.event_name }}" = "push" ] \
+              ( [[ "${{ github.event_name }}" =~ ^(push|workflow_dispatch|schedule) ]] ) \
```

Comment:
can you explain why we need to enclose this into parenthesis and use [[ ]] instead of single ones? Or if we need that why not convert the rest of the conditional too?

thanks (my bash fu is getting old I guess)

Reply:
I've only met this recently, but inside bash's test operator `[[` you can use a regex pattern matching.
I only wanted to avoid a long logical expression checking against 3 types of github event types, so I've simplified it to one regex pattern.

https://www.gnu.org/software/bash/manual/bash.html#Conditional-Constructs

> An additional binary operator, ‘=~’, is available, with the same precedence as ‘==’ and ‘!=’. When it is used, the string to the right of the operator is considered a POSIX extended regular expression and matched accordingly (using the POSIX regcomp and regexec interfaces usually described in regex(3)).

Reply:
I understand [[]] now but what about the parenthesis?

Reply:
that's probably unnecessary at this point: I was trying to solve this in a different way first, which needed the parentheses to group the same way it's already used.
I'll remove it.